### PR TITLE
Benchmarks: microbenchmark - add auto selecting algorithm support for cudnn functions

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
add auto selecting algorithm support for cudnn functions.

**Major Revision**
- add auto selecting algorithm support for cudnn functions in source code
- add 'auto_algo' option in benchmark
- add related test

### CODE REVIEW

**Review by cp5555 — APPROVED** [review]

File: superbench/benchmarks/micro_benchmarks/cudnn_function.py [hunk 1]
```diff
@@ -373,6 +380,8 @@ def _preprocess(self):
         command += (' --warm_up ' + str(self._args.num_warmup))
         command += (' --num_in_step ' + str(self._args.num_in_step))
         command += (' --random_seed ' + str(self._args.random_seed))
+        if self._args.auto_algo:
+            command += (' --auto_algo ')
```
Comment (abuccts) [inline superbench/benchmarks/micro_benchmarks/cudnn_function.py:384]: don't need the last space?

File: superbench/benchmarks/micro_benchmarks/cudnn_function/cudnn_function_helper.h [hunk 2]
```diff
@@ -91,6 +103,7 @@ class Options {
         random_seed = get_cmd_line_argument_int("--random_seed");
         random_seed = (random_seed == 0 ? time(NULL) : random_seed);
         para_info_json = get_cmd_line_argument_string("--config_json");
+        auto_algo = get_cmd_line_argument_bool("--auto_algo");
```
Comment (abuccts) [inline superbench/benchmarks/micro_benchmarks/cudnn_function/cudnn_function_helper.h:106]: maybe enable_auto_algo?

**Review by abuccts — APPROVED** [review]

**Review by guoshzhao — APPROVED** [review]

File: tests/benchmarks/micro_benchmarks/test_cudnn_function.py [hunk 3]
```diff
@@ -126,3 +125,27 @@ def test_cudnn_functions():
         assert (isinstance(benchmark.result[metric][0], numbers.Number))
         if metric != 'return_code':
             assert (len(benchmark.raw_data[metric][0]) == benchmark._args.num_steps)
+
+
+# Test for auto_algo parameter
```
Comment (guoshzhao) [inline tests/benchmarks/micro_benchmarks/test_cudnn_function.py:130]: Alignment.

File: superbench/benchmarks/micro_benchmarks/cudnn_function.py (hunk 1 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cudnn_function.py:384]: yes, removed

File: superbench/benchmarks/micro_benchmarks/cudnn_function/cudnn_function_helper.h (hunk 2 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cudnn_function/cudnn_function_helper.h:106]: ok, revised

File: tests/benchmarks/micro_benchmarks/test_cudnn_function.py (hunk 3 above)
Comment (yukirora) [reply tests/benchmarks/micro_benchmarks/test_cudnn_function.py:130]: thanks, updated

**Pull request merged**